fenix/app/src/main/java/org/mozilla/fenix/gptintegration/apis/GPTIntegrationApi.kt

```diff
@@ -0,0 +1,49 @@
+package org.mozilla.fenix.gptintegration.apis
+
+import mozilla.components.concept.fetch.Client
+import mozilla.components.concept.fetch.MutableHeaders
+import mozilla.components.concept.fetch.Request
+import mozilla.components.concept.fetch.Response
+import org.json.JSONArray
+import org.json.JSONObject
+import org.mozilla.fenix.utils.Settings
+
+private const val BASE_API_URL = "https://api.openai.com/"
+
+class GPTIntegrationApi(
+    settings: Settings,
+    private val client: Client,
+): GPTIntegrationApiInterface {
+
+    private val apiKey = settings.gptApiKey
+
+    override suspend fun fetch(body: String?): Response {
+        val jsonObject = JSONObject()
+        val jsonArray = JSONArray()
+        val jsonObjectSubItem = JSONObject()
+        jsonObjectSubItem.put("role", "user")
+        jsonObjectSubItem.put("content",
+         "Summarize the following text: $body"
+        )
+        jsonArray.put(jsonObjectSubItem)
+        jsonObject.put("model", "gpt-3.5-turbo")
```

Comment:
Improve: If I had to make this code even more flexible, I would add this value into an Enum and expose it on settings/onboarding, so users would be able to decide which version of the model they want to use.